hugo/content/docs/smodel.md

```diff
@@ -0,0 +1,554 @@
+
+*Default features:*
+
+* [editFeature](#editfeature)
+* [deletableFeature](#deletablefeature)
+* [selectFeature](#selectfeature)
+* [fadeFeature](#fadefeature)
+* [hoverFeedbackFeature](#hoverfeedbackfeature)
+
+### SLabelImpl
+
+A label can be attached to a node, port, or edge.
+
+*Properties:*
+
+* `text: string`: The text of the label.
+* `selected: boolean`: Indicates if the label is selected. Defaults to `false`.
+* `alignment: Point`: The alignment of the label. Defaults to `Point.ORIGIN`.
+* `opacity: number`: The opacity of the label. Defaults to `1`.
+* `edgePlacement: EdgePlacement` - *optional*: The placement of the label on an edge. Defaults to `EdgePlacement.NONE`.
+
+*Inheritance:*
+
+[`SShapeElementImpl`](#sshapeelementimpl) &rarr; [`SChildElementImpl`](#schildelementimpl) &rarr; [`SParentElementImpl`](#sparentelementimpl) &rarr; [`SModelElementImpl`](#smodelelementimpl)
+
+*Default features:*
+
+* [boundsFeature](#boundsfeature)
+* [alignFeature](#alignfeature)
+* [layoutableChildFeature](#layoutablechildfeature)
+* [edgeLayoutFeature](#edgelayoutfeature)
+* [fadeFeature](#fadefeature)
+
+### SCompartmentImpl
+
+A compartment is used to group multiple child elements of a node, such as labels. Usually, a `vbox` of `hbox` layout is used to arrange these children.
+
+*Properties*:
+
+* `layout: string` - *optional*: The layout of the compartment.
+* `opacity: number`: The opacity of the compartment. Defaults to `1`.
+
+*Inheritance:*
+
+[`SShapeElementImpl`](#sshapeelementimpl) &rarr; [`SChildElementImpl`](#schildelementimpl) &rarr; [`SParentElementImpl`](#sparentelementimpl) &rarr; [`SModelElementImpl`](#smodelelementimpl)
+
+*Default features:*
+
+* [boundsFeature](#boundsfeature)
+* [layoutContainerFeature](#layoutcontainerfeature)
+* [layoutableChildFeature](#layoutablechildfeature)
+* [fadeFeature](#fadefeature)
+
+### SGraphIndex
+
+A specialized model index that tracks outgoing and incoming edges.
+
+*Inheritance:*
+
+[`ModelIndexImpl`](#modelindeximpl)
+
+### ViewportRootElement
+
+Model root element that defines a viewport, so it transforms the coordinate system of its children with a scroll and zoom factor.
+
+*Properties:*
+
+* `scroll: Point`: The scroll offset of the viewport. Defaults to `{x: 0, y: 0}`.
+* `zoom: number`: The zoom factor of the viewport. Defaults to `1`.
+* `position: Point`: The position of the viewport. Defaults to `{x: 0, y: 0}`.
+* `size: Dimension`: The size of the viewport. Defaults to `{width: -1, height: -1}`.
+
+*Inheritance:*
+[`SModelRootImpl`](#smodelrootimpl) &rarr; [`SParentElementImpl`](#sparentelementimpl) &rarr; [`SModelElementImpl`](#smodelelementimpl)
+
+### SConnectableElementImpl
+
+A connectable element is on that can have outgoing and incoming edges. It can be the source or target element of an edge. There are two kinds of connectable elements: nodes and ports.
```

Comment:
A connectable element is on[e] that can have outgoing and incoming edges. It can be the source or target element of an edge. There are two kinds of connectable elements: [SN]odes and [SP]orts.